common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -507,43 +492,30 @@ def ensure_user_information(
     # invariants have been violated and future misbehavior is likely.
     user_inactive = user and not user.is_active
     user_unset = user is None
-    dispatch_to_login = is_login and (user_unset or user_inactive)
+
+    dispatch_to_login = (
+        ((is_login or is_login_2) and (user_unset or user_inactive))
+        or
+        ((is_register or is_register_2) and user_inactive)
```

Comment:
I think this isn't quite correct.  When the pipeline gets to this point from the registration page, if the user doesn't have an account (`user = None`), then `user_inactive` will be True.  In that case, we still want to send them to the register page; but here it looks like they'll get sent to the login page.